lib/route.js

```diff
@@ -214,7 +220,8 @@ function buildRouting (options) {
 
           buildSchema(context, opts.schemaCompiler || this[kSchemaCompiler], this[kSchemas])
         } catch (error) {
-          done(error)
+          // bubble up the FastifyError instance
+          done(error.code ? error : new FST_ERR_SCH_BUILD(opts.method, url, error.message))
```

Comment:
I think we can add the plugin name to the error when a plugin is used.

Reply:
Good idea!
I will think a better function to define the error message so 👍

Reply:
I have checked to get the plugin name, but when we register the routes it seems that we don't have any reference to the creator of the route.

The only reference to plugins is `this[Symbol.for('registered-plugin')]` that is an array

I was looking for `Symbol.for('plugin-meta')` that contain that information but the function that should have it has been executed.

Do you have any tips to find this function?

Reply:
That's going to be hardish. Avvio looks for a plugin name using this code: https://github.com/mcollina/avvio/blob/master/plugin.js#L9-L28.
It's _hacky_, and I would not add this for the sake of an error.

Reply:
@StarpTech are you ok with my explanation, or would you like this changed nevertheless?

Reply:
As far I remember this logic is only used when no plugin name was given to the fastify plugin metadata right? When the developer has define a name I think we should try to add it to the error object. I can imagine it's very useful for tracing tools.

Reply:
Imagine an error is reported and your code was updated in the meantime. The stack does not point to the correct code but with the plugin name you would have a starting point. Due to the fact that we can't know how people manage their directory structure the plugin name can be really useful.

Reply:
I don't think we keep track of that value anywhere. I think that would be a nice feature to add for a huge number of reasons. Would you be ok if we open an new issue and land this without it? I fear that would be a significant refactoring and this PR is already helping a lot by showing the actual route.

Reply:
Yeah, definitely.